itest/lnd_multi-hop_test.go

```diff
@@ -740,10 +740,283 @@ func runMultiHopRemoteForceCloseOnChainHtlcTimeout(ht *lntest.HarnessTest,
+	// if this is a leased channel. In that case, we'd also check the HTLC
+	// stages are correct in that channel.
+	if c == lnrpc.CommitmentType_SCRIPT_ENFORCED_LEASE {
+		ht.AssertNumPendingForceClose(bob, 2)
+		ht.AssertNumHTLCsAndStage(bob, bobChanPoint, 1, 1)
+	} else {
+		ht.AssertNumPendingForceClose(bob, 1)
+	}
+
+	// Now that the preimage from Bob has hit the chain, restart Alice to
+	// ensure she'll pick it up.
+	require.NoError(ht, restartAlice())
+
+	// If we then mine additional blocks, both Bob and Carol's second level
+	// txes will mature, and they should pull the funds.
+	ht.MineBlocks(carolSecondLevelCSV)
+
+	// We should now see 2 transactions,
+	// 1. Carol's sweeping of her second level tx.
+	// 2. Bob's sweeping of his second level tx.
+	ht.Miner.AssertNumTxsInMempool(2)
+
+	// Mine one more block to confirm both Bob and Carol's sweeps.
+	ht.MineBlocksAndAssertNumTxes(1, 2)
+
+	// With the script-enforced lease commitment type, Alice and Bob still
+	// haven't been able to sweep their respective commit outputs due to
+	// the additional CLTV. We'll need to mine enough blocks for the
+	// timelock to expire and prompt their sweep.
+	if c == lnrpc.CommitmentType_SCRIPT_ENFORCED_LEASE {
+		for _, node := range []*node.HarnessNode{alice, bob} {
+			ht.AssertNumPendingForceClose(node, 1)
+		}
+
+		// Due to the way the test is set up, Alice and Bob share the
+		// same CLTV for their commit outputs even though it's enforced
+		// on different channels (Alice-Bob and Bob-Carol).
+		resp := alice.RPC.PendingChannels()
+		require.Len(ht, resp.PendingForceClosingChannels, 1)
+		forceCloseChan := resp.PendingForceClosingChannels[0]
+		require.Positive(ht, forceCloseChan.BlocksTilMaturity)
+
+		// Mine enough blocks for the timelock to expire.
+		numBlocks := uint32(forceCloseChan.BlocksTilMaturity)
+		ht.MineBlocks(numBlocks)
+
+		// Both Alice and Bob show broadcast their commit sweeps.
+		aliceCommitOutpoint := wire.OutPoint{
+			Hash: *bobForceClose, Index: 3,
+		}
+		aliceCommitSweep := ht.Miner.AssertOutpointInMempool(
+			aliceCommitOutpoint,
+		).TxHash()
+		bobCommitOutpoint := wire.OutPoint{Hash: closingTxid, Index: 3}
+		bobCommitSweep := ht.Miner.AssertOutpointInMempool(
+			bobCommitOutpoint,
+		).TxHash()
+
+		// Confirm their sweeps.
+		block := ht.MineBlocksAndAssertNumTxes(1, 2)[0]
+		ht.Miner.AssertTxInBlock(block, &aliceCommitSweep)
+		ht.Miner.AssertTxInBlock(block, &bobCommitSweep)
+	}
+
+	// All nodes should show zero pending and open channels.
+	for _, node := range []*node.HarnessNode{alice, bob, carol} {
+		ht.AssertNumPendingForceClose(node, 0)
+		ht.AssertNodeNumChannels(node, 0)
+	}
+
+	// Finally, check that the Alice's payment is correctly marked
+	// succeeded.
+	ht.AssertPaymentStatus(alice, preimage, lnrpc.Payment_SUCCEEDED)
 }
 
-func runMultiHopHtlcLocalChainClaim(ht *lntest.HarnessTest,
+// runMultiHopHtlcLocalChainClaimMempool runs the test
```

Comment:
Should be `runMultiHopHtlcLocalChainClaimBlock`

Reply:
fixed!